### PR TITLE
Added Buxfer API integration "storage" following the existing design pattern and style

### PR DESCRIPTION
**Pull request by nissant**
- Added [Buxfer ](https://wwwn.buxfer.com/help/api/) API integration
- Updated README
- Tested on local machine
- Tests pass
- Linter pass

### CODE REVIEW

**Comment by daniel-hauser** [comment]
Thanks @nissant for this change!
Can you go over my changes in the PR and ack you are ok with them?

**Comment by nissant** [comment]
Thanks for those additions @daniel-hauser!
Following some tests on my accounts I will push an additional change to [buxfer-ts-client](https://github.com/nissant/buxfer-ts-client) package and up the minor package version here.
Will ping here once it's ready to merge.

**Comment by nissant** [comment]
@daniel-hauser - Done. Please review and approve.

File: src/storage/buxfer.ts [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+    if (missingAccounts.size > 0) {
+      logger(`Accounts missing in Buxfer_ACCOUNTS:`, missingAccounts);
+    }
+
+    return stats;
+  }
+
+  tagTransactionsByRules(txToSend: BuxferTransaction[]) {
+    // TODO - Implement declarative rule engine
+    let tags: string[] = new Array("added-by-moneyman-etl");
```
Comment (daniel-hauser) [inline src/storage/buxfer.ts]:
```suggestion
    const tags: string[] = ["added-by-moneyman-etl"];
```

File: src/storage/buxfer.ts [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+
+    if (missingAccounts.size > 0) {
+      logger(`Accounts missing in Buxfer_ACCOUNTS:`, missingAccounts);
+    }
+
+    return stats;
+  }
+
+  tagTransactionsByRules(txToSend: BuxferTransaction[]) {
+    // TODO - Implement declarative rule engine
```
Comment (daniel-hauser) [inline src/storage/buxfer.ts:89]: Do you have any idea on how you want to create it?
See #120

File: src/storage/buxfer.ts (hunk 1 above)
Comment (nisan-tagar) [reply src/storage/buxfer.ts]: Thanks committed.

File: src/storage/buxfer.ts (hunk 2 above)
Comment (nisan-tagar) [reply src/storage/buxfer.ts:89]: Well I have not tested or attempted a POC yet, but generally I think its best to have a set of JSON file rules that each user could define for himself **and not committed.** The rules would define the auto tagging logic for each transaction to be uploaded to storages.
I am having a look at some packages that can accomplish this:
https://github.com/CacheControl/json-rules-engine
https://github.com/andrewvo89/rules-engine-ts

**Pull request merged**